### PR TITLE
onnxruntime: 1.24.4 -> 1.26.0

### LINKED ISSUE #523209: Update Request: onnxruntime 1.24.4 → 1.26.0
### Nixpkgs version

- Unstable (26.05)

### Package name

onnxruntime

### Upstream version

1.26.0

### Nixpkgs version

1.24.4

### Changelog

https://github.com/microsoft/onnxruntime/releases/tag/v1.26.0

### Additional context

_No response_

### Notify maintainers

@puffnfresh @ck3d 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this package update does not yet exist in an [open pull request](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+package+%28update%29%22) or in [Nixpkgs Unstable](https://search.nixos.org/packages?channel=unstable).
- [x] I assert that this is not a [duplicate of any known issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%229.needs%3A+package+%28update%29%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ck3d**
update to latest version and fixes #523209 
Additionally this takes over #450107 to fix cross-compilation #306042.

`nix-build -A pkgsCross.aarch64-multiplatform.onnxruntime`

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.
- [x] Follows the [automation/AI policy].

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[automation/AI policy]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

### CODE REVIEW

**Comment by ck3d** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 523244`
Commit: `4b8c9d7df42c652d2dac2824841f3667059ddc48`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python313Packages.torchrl</li>
    <li>vcmi</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 170 packages built:</summary>
  <ul>
    <li>adoptopenjdk-icedtea-web</li>
    <li>aider-chat-full</li>
    <li>aider-chat-with-help</li>
    <li>aitrack</li>
    <li>alpaca</li>
    <li>babeldoc</li>
    <li>calibre</li>
    <li>calibre-no-speech</li>
    <li>colmap</li>
    <li>colmapWithCuda</li>
    <li>convertx</li>
    <li>crewai (python313Packages.crewai)</li>
    <li>crow-translate</li>
    <li>deface</li>
    <li>exo</li>
    <li>firefox-beta-unwrapped</li>
    <li>firefox-devedition-unwrapped</li>
    <li>firefox-unwrapped</li>
    <li>firefoxpwa-unwrapped</li>
    <li>fotema</li>
    <li>frigate</li>
    <li>hieroglyphic</li>
    <li>hns</li>
    <li>hyprwhspr-rs</li>
    <li>icm</li>
    <li>ieda</li>
    <li>immich-machine-learning</li>
    <li>kikoplay</li>
    <li>lacus</li>
    <li>libretranslate (python313Packages.libretranslate)</li>
    <li>librewolf-unwrapped</li>
    <li>livecaptions</li>
    <li>maa-assistant-arknights</li>
    <li>maa-cli</li>
    <li>magika (python313Packages.magika)</li>
    <li>magika-cli</li>
    <li>markitdown-mcp</li>
    <li>minari (python313Packages.minari)</li>
    <li>monado</li>
    <li>newelle</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>onnxruntime</li>
    <li>open-webui</li>
    <li>openseeface</li>
    <li>opentrack</li>
    <li>owocr</li>
    <li>piper-phonemize</li>
    <li>piper-tts</li>
    <li>python313Packages.ale-py</li>
    <li>python313Packages.amd-quark</li>
    <li>python313Packages.argos-translate-files</li>
    <li>python313Packages.argostranslate</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.brax</li>
    <li>python313Packages.chromadb</li>
    <li>python313Packages.cleanlab</li>
    <li>python313Packages.clu</li>
    <li>python313Packages.deepface</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.dm-sonnet</li>
    <li>python313Packages.edward</li>
    <li>python313Packages.evosax</li>
    <li>python313Packages.fastembed</li>
    <li>python313Packages.faster-whisper</li>
    <li>python313Packages.flax</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.gliner</li>
    <li>python313Packages.gymnasium</li>
    <li>python313Packages.helium</li>
    <li>python313Packages.insightface</li>
    <li>python313Packages.keras</li>
    <li>python313Packages.lacuscore</li>
    <li>python313Packages.langchain-chroma</li>
    <li>python313Packages.langchain-huggingface</li>
    <li>python313Packages.lerobot</li>
    <li>python313Packages.llama-index-embeddings-huggingface</li>
    <li>python313Packages.llama-index-vector-stores-chroma</li>
    <li>python313Packages.llm-sentence-transformers</li>
    <li>python313Packages.manifest-ml</li>
    <li>python313Packages.markitdown</li>
    <li>python313Packages.minisbd</li>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mtcnn</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.onnx-asr</li>
    <li>python313Packages.onnx-ir</li>
    <li>python313Packages.onnxruntime</li>
    <li>python313Packages.onnxscript</li>
    <li>python313Packages.pettingzoo</li>
    <li>python313Packages.piper-phonemize</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pypasser</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.rapidocr</li>
    <li>python313Packages.rapidocr-onnxruntime</li>
    <li>python313Packages.rembg</li>
    <li>python313Packages.retinaface</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.sentence-transformers</li>
    <li>python313Packages.sherpa-onnx</li>
    <li>python313Packages.shimmy</li>
    <li>python313Packages.skrl</li>
    <li>python313Packages.speechrecognition</li>
    <li>python313Packages.stable-baselines3</li>
    <li>python313Packages.tami4edgeapi</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tf2onnx</li>
    <li>python313Packages.tianshou</li>
    <li>python313Packages.tinygrad</li>
    <li>python313Packages.translatehtml</li>
    <li>python313Packages.txtai</li>
    <li>ultralytics (python313Packages.ultralytics)</li>
    <li>python313Packages.unstructured-inference</li>
    <li>python313Packages.vmas</li>
    <li>python313Packages.vqgan-jax</li>
    <li>python313Packages.waymax</li>
    <li>whisperx (python313Packages.whisperx)</li>
    <li>python314Packages.amd-quark</li>
    <li>python314Packages.chromadb</li>
    <li>python314Packages.crewai</li>
    <li>python314Packages.fastembed</li>
    <li>python314Packages.faster-whisper</li>
    <li>python314Packages.gliner</li>
    <li>python314Packages.helium</li>
    <li>python314Packages.insightface</li>
    <li>python314Packages.lacuscore</li>
    <li>python314Packages.langchain-chroma</li>
    <li>python314Packages.langchain-huggingface</li>
    <li>python314Packages.llm-sentence-transformers</li>
    <li>python314Packages.magika</li>
    <li>python314Packages.manifest-ml</li>
    <li>python314Packages.markitdown</li>
    <li>python314Packages.minisbd</li>
    <li>python314Packages.mmcv</li>
    <li>python314Packages.onnx-asr</li>
    <li>python314Packages.onnx-ir</li>
    <li>python314Packages.onnxruntime</li>
    <li>python314Packages.onnxscript</li>
    <li>python314Packages.piper-phonemize</li>
    <li>python314Packages.pypasser</li>
    <li>python314Packages.rapidocr</li>
    <li>python314Packages.rapidocr-onnxruntime</li>
    <li>python314Packages.rembg</li>
    <li>python314Packages.sentence-transformers</li>
    <li>python314Packages.sherpa-onnx</li>
    <li>python314Packages.speechrecognition</li>
    <li>python314Packages.tami4edgeapi</li>
    <li>python314Packages.tinygrad</li>
    <li>python314Packages.txtai</li>
    <li>python314Packages.ultralytics</li>
    <li>python314Packages.unstructured-inference</li>
    <li>python314Packages.whisperx</li>
    <li>rabbit-ng</li>
    <li>rapidraw</li>
    <li>rembg</li>
    <li>seagoat</li>
    <li>sherpa-onnx</li>
    <li>sitespeed-io</li>
    <li>tests.home-assistant-components.tami4</li>
    <li>thunderbird-unwrapped (thunderbird-latest-unwrapped, thunderbirdPackages.thunderbird, thunderbirdPackages.thunderbird-latest)</li>
    <li>unbook</li>
    <li>unstructured-api</li>
    <li>vectorcode</li>
    <li>vimPlugins.vectorcode-nvim</li>
    <li>voxtype-onnx</li>
    <li>vtk_9_6</li>
    <li>whisper-ctranslate2</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-piper</li>
  </ul>
</details>

File: pkgs/by-name/on/onnxruntime/package.nix [hunk 1]
```diff
@@ -269,7 +269,7 @@ effectiveStdenv.mkDerivation (finalAttrs: {
   cmakeFlags = [
     (lib.cmakeBool "ABSL_ENABLE_INSTALL" true)
     # leads to failing builds, which isn't particularly useful for Nixpkgs
-    (lib.cmakeFeature "CMAKE_CXX_FLAGS" "-Wno-error=unused-variable")
+    (lib.cmakeFeature "CMAKE_CXX_FLAGS" "-Wno-error=unused-variable -Wno-error=deprecated -Wno-error=unused-result -Wno-error=sign-compare")
```
Comment (SuperSandro2000) [inline pkgs/by-name/on/onnxruntime/package.nix]:
```suggestion
    (lib.cmakeFeature "CMAKE_CXX_FLAGS" "-Wno-error")
```
How about just doing this?

Comment (ck3d) [reply pkgs/by-name/on/onnxruntime/package.nix]: It did not work, but I could reduce the list a little bit.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

File: pkgs/by-name/on/onnxruntime/package.nix
Comment (pcc) [inline pkgs/by-name/on/onnxruntime/package.nix:296]: Do we need to set `Python_NumPy_INCLUDE_DIR`  here as in #450107?

Comment (ck3d) [reply pkgs/by-name/on/onnxruntime/package.nix:296]: Python support is disabled for cross compilation. I tried to enable it, but the dependency openblas faild to cross-compile. I think we should open another PR to support also cross compilation of the Python package.

Comment (pcc) [reply pkgs/by-name/on/onnxruntime/package.nix:296]: Got it. #450106 was supposed to fix it but I guess it's broken again.

**Review by pcc — APPROVED** [review]
